gateway-ha/src/main/java/io/trino/gateway/ha/config/OAuth2GatewayCookieProperties.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.config;
+
+import io.airlift.units.Duration;
+
+import java.util.List;
+
+public class OAuth2GatewayCookieProperties
+{
+    private static final OAuth2GatewayCookieProperties instance = new OAuth2GatewayCookieProperties();
+
+    private OAuth2GatewayCookieConfiguration oAuth2GatewayCookieConfiguration;
+
+    private OAuth2GatewayCookieProperties()
+    {}
+
+    public static OAuth2GatewayCookieProperties getInstance()
+    {
+        return instance;
+    }
+
+    public void initialize(OAuth2GatewayCookieConfiguration oAuth2GatewayCookieConfiguration)
+    {
+        this.oAuth2GatewayCookieConfiguration = oAuth2GatewayCookieConfiguration;
+    }
+
+    public List<String> getDeletePaths()
+    {
+        ensureInitialized();
+        return oAuth2GatewayCookieConfiguration.getDeletePaths();
+    }
+
+    public List<String> getRoutingPaths()
+    {
+        ensureInitialized();
+        return oAuth2GatewayCookieConfiguration.getRoutingPaths();
+    }
+
+    public Duration getLifetime()
+    {
+        ensureInitialized();
+        return oAuth2GatewayCookieConfiguration.getLifetime();
+    }
+
+    private void ensureInitialized()
+    {
+        if (oAuth2GatewayCookieConfiguration == null) {
+            throw new IllegalStateException("GatewayCookieConfiguration not initialized before use");
```

Comment:
this should be `oAuth2GatewayCookieConfiguration` ?
Btw in which cases would oAuth2GatewayCookieConfiguration not be initialized?

Reply:
It is initialized during startup, so this error shouldn't be reachable unless we introduce a bug.

Reply:
>this should be oAuth2GatewayCookieConfiguration ?
can you clarify?

Reply:
Oh i meant the message `"GatewayCookieConfiguration not initialized before use"` shouldn't it be `"oAuth2GatewayCookieConfiguration not initialized before use"`

Reply:
Ah yes, thanks! I added some extra guidance to make the error message more descriptive